Pod/VOKNavigationHelper/VOKNavigationHelper.m

```diff
@@ -0,0 +1,139 @@
+//
+//  VOKNavigationHelper.m
+//  VOKUtilities
+//
+//  Created by Brock Boland on 1/24/17.
+//  Copyright (c) 2017 Vokal. All rights reserved.
+//
+
+#import "VOKNavigationHelper.h"
+
+@implementation VOKNavigationHelper
+
++ (void)clearExistingViewsAndAnimateToViewController:(UIViewController *)viewController
+                                            duration:(NSTimeInterval)duration
+                                             options:(UIViewAnimationOptions)options
+                                          completion:(VOKViewAnimationCompletionBlock)completion
+{
+    // Enforce the non-null
+    NSParameterAssert(viewController != nil);
+    if (viewController == nil) {
+        return;
+    }
+    
+    // Helper block: execute the completion block (if there is one) on the main thread
+    void (^executeCompletionOnMainThread)() = ^void() {
+        if (completion) {
+            [[NSOperationQueue mainQueue] addOperationWithBlock:^{
+                completion(YES);
+            }];
+        }
+    };
+
+    // Find the window in which this is all taking place
+    id<UIApplicationDelegate> appDelegate = UIApplication.sharedApplication.delegate;
+    
+    if (!appDelegate.window) {
+        appDelegate.window = [[UIWindow alloc] initWithFrame:UIScreen.mainScreen.bounds];
+        appDelegate.window.screen = UIScreen.mainScreen;
+        [appDelegate.window makeKeyAndVisible];
+    }
+    UIWindow *appWindow = appDelegate.window;
+
+    // Snapshot the current view, if the transition will be animated
+    UIView *snapshot;
+    if (duration > 0) {
+        snapshot = [UIScreen.mainScreen snapshotViewAfterScreenUpdates:NO];
+    }
+    
+    // Helper block: replace the root view controller, and animate the transition if desired
+    void (^switchToNewViewController)() = ^void() {
+        appWindow.rootViewController = viewController;
+        if (duration > 0 && snapshot != nil) {
+            [appWindow addSubview:snapshot];
+            [UIView transitionFromView:snapshot
+                                toView:viewController.view
+                              duration:duration
+                               options:options
+                            completion:^(BOOL finished) {
+                                [snapshot removeFromSuperview];
+                                executeCompletionOnMainThread();
+                            }];
+        } else {
+            executeCompletionOnMainThread();
+        }
+    };
+
+    // Get rid of old VC's and views
+    [appWindow.subviews makeObjectsPerformSelector:@selector(removeFromSuperview)];
+
+    // Dismiss any presented view controllers, then [potentially] animate in the new VC
+    if (appWindow.rootViewController.presentedViewController) {
+        [appWindow.rootViewController dismissViewControllerAnimated:NO
```

Comment:
I don't think we'd want to animate this dismissal, and *then* animate to the new view controller—it should just animate from whatever is visible to the new view controller.

I'm sure we could come up with a case in which that wouldn't be the desired outcome, but for the general case that we're addressing here, I think this solution makes sense.